## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 467073`
Commit: [`f483522ddd5efe5eb5e6b5c78a1be5d794ecd4d3`](https://github.com/NixOS/nixpkgs/commit/f483522ddd5efe5eb5e6b5c78a1be5d794ecd4d3) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/f483522ddd5efe5eb5e6b5c78a1be5d794ecd4d3..pull/467073/head))
Merge: [`8ea4b5e72ea05727ce2321a1f4f49f5fad5631a3`](https://github.com/NixOS/nixpkgs/commit/8ea4b5e72ea05727ce2321a1f4f49f5fad5631a3)

Logs: https://github.com/liberodark/nixpkgs-review-gha/actions/runs/19854569488


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>timr-tui</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>timr-tui</li>
  </ul>
</details>

---
### `x86_64-darwin` (sandbox = relaxed)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>timr-tui</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>timr-tui</summary>
<pre>  CFLAGS_x86_64-apple-darwin = None
  CARGO_ENCODED_RUSTFLAGS = Some()
  cargo:warning=Warning: supplying the --target x86_64-apple-macosx != x86_64-apple-darwin argument to a nix-wrapped compiler may not work correctly - cc-wrapper is currently not designed with multi-target compilers in mind. You may want to use an un-wrapped compiler instead.
  cargo:warning=error: unable to open output file &#x27;/homeless-shelter/.cache/clang/ModuleCache/306GO983KYHPH/Cocoa-3DWB21RBSB0IU.pcm&#x27;: &#x27;No such file or directory&#x27;
  cargo:warning=In file included from objc/notify.m:1:
  cargo:warning=objc/notify.h:1:9: fatal error: could not build module &#x27;Cocoa&#x27;
  cargo:warning=    1 | #import &lt;Cocoa/Cocoa.h&gt;
  cargo:warning=      |  ~~~~~~~^
  cargo:warning=error: unable to open output file &#x27;/homeless-shelter/.cache/clang/ModuleCache/306GO983KYHPH/CoreServices-1RKWG08JTU0RG.pcm&#x27;: &#x27;No such file or directory&#x27;
  cargo:warning=error: unable to open output file &#x27;/homeless-shelter/.cache/clang/ModuleCache/306GO983KYHPH/Foundation-DSYWQYGMEXBE.pcm&#x27;: &#x27;No such file or directory&#x27;
  cargo:warning=error: unable to open output file &#x27;/homeless-shelter/.cache/clang/ModuleCache/306GO983KYHPH/ObjectiveC-B7GSK2KKDDF6.pcm&#x27;: &#x27;No such file or directory&#x27;
  cargo:warning=5 errors generated.

  --- stderr


  error occurred in cc-rs: command did not execute successfully (status code exit status: 1): env -u IPHONEOS_DEPLOYMENT_TARGET LC_ALL=&quot;C&quot; &quot;/nix/store/fvj17s7clsf2n4rhyq0qqcrqhzybdqrz-clang-wrapper-21.1.2/bin/cc&quot; &quot;-O3&quot; &quot;-ffunction-sections&quot; &quot;-fdata-sections&quot; &quot;-fPIC&quot; &quot;-m64&quot; &quot;--target=x86_64-apple-macosx&quot; &quot;-mmacosx-version-min=14.0&quot; &quot;-Wall&quot; &quot;-Wextra&quot; &quot;-fmodules&quot; &quot;-Wno-deprecated-declarations&quot; &quot;-mmacos-version-min=14.0&quot; &quot;-o&quot; &quot;/nix/build/nix-9005-423295345/source/target/x86_64-apple-darwin/release/build/mac-notification-sys-fcf50e7b688f0506/out/54532c69c7c8f551-notify.o&quot; &quot;-c&quot; &quot;objc/notify.m&quot;


warning: build failed, waiting for other jobs to finish...</pre>
</details>
</details>

---
### `aarch64-darwin` (sandbox = relaxed)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>timr-tui</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>timr-tui</summary>
<pre>  CFLAGS_aarch64-apple-darwin = None
  CARGO_ENCODED_RUSTFLAGS = Some()
  cargo:warning=Warning: supplying the --target arm64-apple-macosx != arm64-apple-darwin argument to a nix-wrapped compiler may not work correctly - cc-wrapper is currently not designed with multi-target compilers in mind. You may want to use an un-wrapped compiler instead.
  cargo:warning=error: unable to open output file &#x27;/homeless-shelter/.cache/clang/ModuleCache/WL5FSVGIUTY3/Cocoa-20ERRVI91BVYU.pcm&#x27;: &#x27;No such file or directory&#x27;
  cargo:warning=In file included from objc/notify.m:1:
  cargo:warning=objc/notify.h:1:9: fatal error: could not build module &#x27;Cocoa&#x27;
  cargo:warning=    1 | #import &lt;Cocoa/Cocoa.h&gt;
  cargo:warning=      |  ~~~~~~~^
  cargo:warning=error: unable to open output file &#x27;/homeless-shelter/.cache/clang/ModuleCache/WL5FSVGIUTY3/CoreServices-99DQKDFMC48K.pcm&#x27;: &#x27;No such file or directory&#x27;
  cargo:warning=error: unable to open output file &#x27;/homeless-shelter/.cache/clang/ModuleCache/WL5FSVGIUTY3/Foundation-1TLYSVBKJUW2T.pcm&#x27;: &#x27;No such file or directory&#x27;
  cargo:warning=error: unable to open output file &#x27;/homeless-shelter/.cache/clang/ModuleCache/WL5FSVGIUTY3/ObjectiveC-285213M0WRDFF.pcm&#x27;: &#x27;No such file or directory&#x27;
  cargo:warning=5 errors generated.

  --- stderr


  error occurred in cc-rs: command did not execute successfully (status code exit status: 1): env -u IPHONEOS_DEPLOYMENT_TARGET LC_ALL=&quot;C&quot; &quot;/nix/store/qk20nysrc2170f1mal5k6r0axqn6jmfj-clang-wrapper-21.1.2/bin/cc&quot; &quot;-O3&quot; &quot;-ffunction-sections&quot; &quot;-fdata-sections&quot; &quot;-fPIC&quot; &quot;--target=arm64-apple-macosx&quot; &quot;-mmacosx-version-min=14.0&quot; &quot;-Wall&quot; &quot;-Wextra&quot; &quot;-fmodules&quot; &quot;-Wno-deprecated-declarations&quot; &quot;-mmacos-version-min=14.0&quot; &quot;-o&quot; &quot;/nix/build/nix-5429-2823600244/source/target/aarch64-apple-darwin/release/build/mac-notification-sys-e64bf1c0fb702ebc/out/54532c69c7c8f551-notify.o&quot; &quot;-c&quot; &quot;objc/notify.m&quot;


warning: build failed, waiting for other jobs to finish...</pre>
</details>
</details>

--- a/pkgs/by-name/ti/timr-tui/package.nix
+++ b/pkgs/by-name/ti/timr-tui/package.nix
@@ -1,0 +1,66 @@
+{
+  lib,
+  stdenv,
+  rustPlatform,
+  fetchFromGitHub,
+  pkg-config,
+  alsa-lib-with-plugins,
+  alsa-plugins,
+  pipewire,
+  writableTmpDirAsHomeHook,
+  versionCheckHook,
+  nix-update-script,
+  enableSound ? false,
+}:
+rustPlatform.buildRustPackage (finalAttrs: {
+  pname = "timr-tui";
+  version = "1.6.1";
+
+  src = fetchFromGitHub {
+    owner = "sectore";
+    repo = "timr-tui";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-s2FnMwDq4tYBaWaT9y1GLbVGFs7zSnOmjcF5leO12JE=";
+  };
+
+  cargoHash = "sha256-9yd348QGjFxt+QmEBuYzd612mFm/PyETrZy4z5wW+nI=";
+
+  # Enable upstream "sound" feature when requested
+  buildFeatures = lib.optionals enableSound [ "sound" ];
+
+  nativeBuildInputs = lib.optionals (enableSound && stdenv.isLinux) [ pkg-config ];
+
+  # Runtime/FFI deps for the sound feature (Linux)
+  buildInputs = lib.optionals (enableSound && stdenv.isLinux) [
+    (alsa-lib-with-plugins.override {
+      plugins = [
+        alsa-plugins
+        pipewire
+      ];
+    })
+  ];
+
+  nativeCheckInputs = [
+    writableTmpDirAsHomeHook
+  ];
+
+  nativeInstallCheckInputs = [
+    versionCheckHook
+  ];
+  versionCheckProgramArg = "--version";
+  # Error: Operation not permitted (os error 1)
+  versionCheckKeepEnvironment = lib.optionals stdenv.hostPlatform.isDarwin [ "HOME" ];
+  doInstallCheck = true;
+
+  passthru.updateScript = nix-update-script { };
+
+  meta = {
+    description = "TUI to organize your time: Pomodoro, Countdown, Timer";
+    homepage = "https://github.com/sectore/timr-tui";
+    changelog = "https://github.com/sectore/timr-tui/releases/tag/v${finalAttrs.version}";
+    license = lib.licenses.mit;
+    mainProgram = "timr-tui";
+    maintainers = [ lib.maintainers.flokkq ];
+    platforms = lib.platforms.unix;
+  };
+})

